# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/MRazaRashid/FlyRank_Week1/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing Lane 2: Refresh / Content Opportunity Scoring. As someone new to applied ML, I want a lane where I can learn end to end workflow (baseline, model, validation, ranked output).

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: Which content pages should a content/SEO reviewer prioritize for refresh, expansion, protection, or pruning, given limited review time?
Who acts, and how: A content strategist or SEO reviewer at a FlyRank client, who has capacity to manually review only a small number of pages (e.g. the top 20-50) each cycle. They use the ranked list to decide where to spend that time first, instead of reviewing pages in an arbitrary or alphabetical order.
Cost of a wrong call: If the model surfaces a page that isn't actually worth reviewing (a false positive), the reviewer wastes limited time on a low-value page while a genuinely declining or high-opportunity page goes unreviewed. If it misses a real opportunity (a false negative), a page that's actually losing visibility or leaving clicks on the table keeps quietly underperforming, unnoticed, until the next review cycle.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:

import os, sys, subprocess, pandas as pd
IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/flyrank-bih/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)
    subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)
elif os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")  # move from notebooks/ to the repo root

print("Working dir:", os.getcwd())
assert os.path.exists("data/raw/content_refresh_anonymized.csv"), "starter CSV not found — are you at the repo root?"
print("Starter data found. You're ready.")

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

df = df[(df["impressions_90d"] > 0) & (df["content_age_days"] >= 90)]
df = df.drop_duplicates(subset="content_id")

print("Total eligible content items:", len(df))
print("Pages currently declining (trend_direction == 'down'):",
      (df["trend_direction"] == "down").sum())
print("Pages that are both declining AND getting real demand "
      "(impressions_90d >= 100):",
      ((df["trend_direction"] == "down") & (df["impressions_90d"] >= 100)).sum())

Working dir: /content/flyrank-ml-internship-starter/flyrank-ml-internship-starter
Starter data found. You're ready.
Total eligible content items: 30000
Pages currently declining (trend_direction == 'down'): 16262
Pages that are both declining AND getting real demand (impressions_90d >= 100): 13152


Out of 30,000 eligible content items, 16,262 (about 54%) are currently flagged as declining. Of those, 13,152 (about 81% of the declining pages) have meaningful search demand (100+ impressions in the last 90 days) — meaning most of the decline isn't happening on dead, low-traffic pages, it's happening on pages people are still actually finding. That's a large, high-signal pool to build and validate a ranking system against, and it's exactly the kind of problem where prioritizing which page to review first has real value.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I can claim: which pages observationally show signs of decline, thinness, or missed CTR opportunity relative to similar pages; a ranked, evidence-backed priority order for review; directional patterns in which signals associate with these opportunities.
What I cannot claim: that refreshing a flagged page will cause it to recover (that requires a controlled experiment, which this data doesn't provide); that any result reflects Google's actual ranking algorithm; that a client, page, or query in this dataset corresponds to any real, identifiable entity. All findings are decision-support for a human reviewer, not a guarantee.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.